code/datums/antagonists/datum_wizard.dm

```diff
@@ -0,0 +1,185 @@
+	to_chat(owner, "The spellbook is bound to you, and others cannot use it.")
+	to_chat(owner, "In your pockets you will find a teleport scroll. Use it as needed.")
+	owner.store_memory("<B>Remember:</B> do not forget to prepare your spells.")
+	SSticker.mode.update_wiz_icons_added(owner)
+	return
+
+/datum/antagonist/wizard/greet()
+	to_chat(owner, "<span class='boldannounce'>You are the Space Wizard!</span>")
+	to_chat(owner, "<B>The Space Wizards Federation has given you the following tasks:</B>")
+	owner.announce_objectives()
+	return
+
+/datum/antagonist/wizard/proc/name_wizard()
+	var/randomname = "[pick(GLOB.wizard_first)] [pick(GLOB.wizard_second)]"
+	var/newname = copytext(sanitize(input(owner.current, "You are the Space Wizard. Would you like to change your name to something else?", "Name change", randomname) as null|text),1,MAX_NAME_LEN)
+	var/mob/living/carbon/human/H = owner.current
+	if(!newname)
+		newname = randomname
+	H.real_name = newname
+	H.name = newname
+	owner.name = newname
+	if(H.age < WIZARD_AGE_MIN)
+		H.age = WIZARD_AGE_MIN
+	H.dna.update_dna_identity()
+	return
+
+/datum/antagonist/wizard/apprentice/on_gain()
+	if(!ishuman(owner.current))
+		LAZYREMOVE(owner.antag_datums, src)
+		qdel(src)
+	SSticker.mode.apprentices += owner
+	owner.special_role = special_role
+	name_wizard()
+	greet()
+	forge_wizard_objectives()
+	finalize_wizard()
+	return
+
+/datum/antagonist/wizard/apprentice/apply_innate_effects()
+	var/mob/living/carbon/human/H = owner.current
+	var/outfit = pick(/datum/outfit/wizard,/datum/outfit/wizard/red,/datum/outfit/wizard/weeb)
+	H.equipOutfit(outfit, TRUE)
+	return
+
+/datum/antagonist/wizard/apprentice/name_wizard()
+	var/randomname = "[pick(GLOB.wizard_first)] [pick(GLOB.wizard_second)]"
+	var/newname = copytext(sanitize(input(owner.current, "You are [summoner.real_name]'s apprentice. Would you like to change your name to something else?", "Name change", randomname) as null|text),1,MAX_NAME_LEN)
+	var/mob/living/carbon/human/H = owner.current
+	if(!newname)
+		newname = randomname
+	H.real_name = newname
+	H.name = newname
+	owner.name = newname
+	H.age = rand(AGE_MIN, WIZARD_AGE_MIN - 1)
+	H.dna.update_dna_identity()
+	return
+
+/datum/antagonist/wizard/apprentice/forge_wizard_objectives()
+	if(summoner)
+		var/datum/objective/protect/new_objective = new /datum/objective/protect
+		new_objective.owner = owner
+		new_objective.target = summoner.mind
+		new_objective.explanation_text = "Protect [summoner.real_name], the wizard."
+		add_objective(new_objective)
+	else
+		var/datum/objective/escape/escape_objective = new
+		escape_objective.owner = owner
+		add_objective(escape_objective)
+	return
+
+/datum/antagonist/wizard/apprentice/greet()
+	if(summoner)
+		to_chat(owner, "<B>You are [summoner.real_name]'s apprentice! You are bound by magic contract to follow their orders and help them in accomplishing their goals.</B>")
```

Comment:
`owner` is a mind, right? Does sending a message to the mind actually do anything? I think it has to be the mob.